binn/importance_network.py

```diff
@@ -279,50 +282,37 @@ def get_fan_out(self, query_node: str):
-
-        - 'nodes_in_subgraph': the total number of nodes in the upstream and downstream subgraphs for each node
+        Adds normalization to the importance values based on the specified method.
 
-        - 'log(nodes_in_subgraph)': the logarithm (base 2) of 'nodes_in_subgraph'
-
-        - 'weighted_val_log': the 'value' column normalized by the logarithm of 'nodes_in_subgraph'
+        Args:
+            method (str): The normalization method to use. Options are "fan" and "subgraph".
+                "fan" normalizes based on fan-in and fan-out values.
+                "subgraph" normalizes based on the number of nodes in the upstream and
+                downstream subgraphs.
 
         Returns:
-            pd.DataFrame:
-                The input dataframe with the added normalization columns.
+            pd.DataFrame: The importance dataframe with the normalized values added.
         """
-        self.importance_df["fan_in"] = self.importance_df.apply(
-            lambda x: self.get_fan_in(x["source"]), axis=1
-        )
-        self.importance_df["fan_out"] = self.importance_df.apply(
-            lambda x: self.get_fan_out(x["source"]), axis=1
-        )
-        self.importance_df["fan_tot"] = self.importance_df.apply(
-            lambda x: x["fan_in"] + x["fan_out"], axis=1)
-        self.importance_df["nodes_in_upstream"] = self.importance_df.apply(
-            lambda x: self.get_nr_nodes_in_upstream_subgraph(x["source"]), axis=1
-        )
-        self.importance_df["nodes_in_downstream"] = self.importance_df.apply(
-            lambda x: self.get_nr_nodes_in_downstream_subgraph(x["source"]), axis=1
-        )
-        self.importance_df["nodes_in_subgraph"] = (
-            self.importance_df["nodes_in_downstream"] +
-            self.importance_df["nodes_in_upstream"]
-        )
-        self.importance_df["log(nodes_in_subgraph)"] = np.log2(
-            self.importance_df["nodes_in_subgraph"])
-        self.importance_df["weighted_val_log"] = self.importance_df.apply(
-            lambda x: x["value"] / (np.log2(x["nodes_in_subgraph"])), axis=1
-        )
+        if method == "fan":
+            fan_in = np.array([self.get_fan_in(x) for x in self.importance["source"]])
+            fan_out = np.array([self.get_fan_out(x) for x in self.importance["source"]])
+            nr_tot = fan_in + fan_out
+        if method == "subgraph":
+            upstream_nodes = np.array(
+                [
+                    self.get_nr_nodes_in_upstream_subgraph(x)
+                    for x in self.importance_df["source"]
+                ]
+            )
+            downstream_nodes = np.array(
+                [
+                    self.get_nr_nodes_in_downstream_subgraph(x)
+                    for x in self.importance_df["source"]
+                ]
+            )
+            nr_tot = upstream_nodes + downstream_nodes
+
+        self.importance_df["value"] = np.log2(nr_tot)
```

Comment:
This looks like it replaces "value" with just the log of the counted nodes ```np.log2(nr_tot)``` before the behavior was to set a ```weighted_val_log``` column by dividing by the normalization factor

```python
        self.importance_df["weighted_val_log"] = self.importance_df.apply(
            lambda x: x["value"] / (np.log2(x["nodes_in_subgraph"])), axis=1
        )
```  

If we are just overwriting the "value" column here, it should be:

```python
self.importance_df["value"] = self.importance_df["value"] / np.log2(nr_tot)
```

Reply:
Yes, completely right. Should be `self.importance_df["value"] = self.importance_df["value"] / np.log2(nr_tot)`